bigquery/google/cloud/bigquery/table.py

```diff
@@ -1099,3 +1148,109 @@ def to_dataframe(self):
         rows = [row.values() for row in iter(self)]
 
         return pandas.DataFrame(rows, columns=column_headers)
+
+
+class TimePartitioningType(object):
+    """Specifies the type of time partitioning to perform."""
+
+    DAY = 'DAY'
+    """str: Generates one partition per day."""
+
+
+class TimePartitioning(object):
+    """Configures time-based partitioning for a table.
+
+    Args:
+        partitioning_type (google.cloud.bigquery.table.TimePartitioningType):
+            The only type supported is
+            :attr:`~google.cloud.bigquery.table.TimePartitioningType.DAY`,
+            which will generate one partition per day.
+        field (str, optional):
+            If set, the table is partitioned by this field. If not set, the
+            table is partitioned by pseudo column ``_PARTITIONTIME``. The field
+            must be a top-level ``TIMESTAMP`` or ``DATE`` field. Its mode must
+            be ``NULLABLE`` or ``REQUIRED``.
+        expiration_ms(int, optional):
+            Number of milliseconds for which to keep the storage for a
+            partition.
+    """
+    def __init__(self, partitioning_type, field=None, expiration_ms=None):
+        self._properties = {}
+        self.partitioning_type = partitioning_type
+        if field is not None:
+            self.field = field
+        if expiration_ms is not None:
+            self.expiration_ms = expiration_ms
+
+    @property
+    def partitioning_type(self):
```

Comment:
Nit: since "partitioning" is redundant with the class name, we could call this `type_` (trailing underscore to prevent conflicts with built-in) See: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4658#issue-284349635